### PR TITLE
Update Jupyter charms to latest upstream version 

### PR DESCRIPTION
**Pull request by knkski**
Also adds functional UI testing to ensure we don't get dashboard regressions

### CODE REVIEW

File: charms/jupyter-controller/reactive/jupyter_controller.py [hunk 1]
```diff
@@ -63,7 +78,8 @@ def start_charm():
                     'name': 'jupyter-controller',
                     'command': ['/manager'],
                     'config': {
-                        'USE_ISTIO': 'false',
+                        'USE_ISTIO': hookenv.is_relation_made('service-mesh'),
```
Comment (johnsca) [inline charms/jupyter-controller/reactive/jupyter_controller.py:81]: The type here is changing from a string to a bool.  I know this is going to be YAML-ified and then probably cast to a string anyway, but just wanted to double-check that there won't be a capitalization issue introduced here?

File: tests/test_selenium.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+from selenium.webdriver.firefox.options import Options
+from selenium.webdriver.support.ui import WebDriverWait
+
+
+def polymer_is_fucking_trash(elems):
+    """Workaround for web components breaking fucking querySelector.
+
+    Who the fuck thought it was a good idea to just yeet the moral equivalent
+    of iframes everywhere over a single page?
+    """
```
Comment (johnsca) [inline tests/test_selenium.py:16]: :rofl:

**Review by johnsca — APPROVED** [review]
LGTM, and +1 to the Selenium test (especially that helper function; I know how annoying those types of tests can be).  One minor question but I'm pretty sure I already know the answer which is that it's not an issue.

File: charms/jupyter-controller/reactive/jupyter_controller.py (hunk 1 above)
Comment (knkski) [reply charms/jupyter-controller/reactive/jupyter_controller.py]: Good catch, that might cause issues:

https://github.com/kubeflow/kubeflow/blob/19da809/components/notebook-controller/controllers/notebook_controller.go#L180

Forcing it to be a lower-case string just to be sure

File: bundle.yaml [hunk 3]
```diff
@@ -3,8 +3,8 @@ applications:
   argo-controller:              { charm: cs:argo-controller,             scale: 1, annotations: { gui-x:  '600', gui-y:    '0' } }
   argo-ui:                      { charm: cs:argo-ui,                     scale: 1, annotations: { gui-x:  '300', gui-y:    '0' } }
   dex-auth:                     { charm: cs:dex-auth,                    scale: 1, annotations: { gui-x:  '600', gui-y: '-518' } }
-  istio-ingressgateway:         { charm: cs:istio-ingressgateway-7,      scale: 1, annotations: { gui-x:    '0', gui-y:    '0' } }
-  istio-pilot:                  { charm: cs:istio-pilot-7,               scale: 1, annotations: { gui-x:    '0', gui-y:    '0' } }
+  istio-ingressgateway:         { charm: cs:istio-ingressgateway-8,      scale: 1, annotations: { gui-x:    '0', gui-y:    '0' } }
```
Comment (DomFleischmann) [inline bundle.yaml:6]: Shouldn't this be done for bundle-lite and bundle-edge also?

File: charms/jupyter-controller/metadata.yaml [hunk 4]
```diff
@@ -10,7 +10,10 @@ resources:
     type: oci-image
     description: 'Backing OCI image'
     auto-fetch: true
-    upstream-source: gcr.io/kubeflow-images-public/notebook-controller:v20191007-v0.6.0-rc.0-134-g6ddf28cc-dirty-560d95
+    upstream-source: gcr.io/kubeflow-images-public/notebook-controller:vmaster-g6eb007d0
+requires:
+  service-mesh:
```
Comment (DomFleischmann) [inline charms/jupyter-controller/metadata.yaml:15]: Shouldn't this relation also be added to the bundle?

File: bundle.yaml (hunk 3 above)
Comment (knkski) [reply bundle.yaml:6]: Whoops, added to bundle-lite. bundle-edge doesn't have the dashboard included in it, though

**Review by johnsca — APPROVED** [review]

**Pull request merged**